### ISSUE TITLE
[QUESTION] Shifted / delayed / lagged prediction results

### ISSUE THREAD

**Issue by nikobenho** [unit8co/darts#1142]
Hi,

Thank you for developing this great package. It's provided me with a smooth introduction to working with neural networks.

I have a weird issue that I couldn't find mentioned anywhere else.

My issue is that my model produces near perfect predictions **but with a lag (or delay) that is == output_chunk_length**.

Please have a look at the screenshot below, where red arrows highlight peaks and troughs in the data that exemplify the lag between the input data and the prediction. On the second image, I've shifted the input data forward by output_chunk_length, and one can see that the fit is significantly improved.

![model_issue_2](https://user-images.githubusercontent.com/46219250/184249627-20972b92-0b1f-4659-a72b-1ab0e6ab091d.PNG)

![model_issue_2_shifted](https://user-images.githubusercontent.com/46219250/184249687-095b81a8-0915-43bd-b9fc-0bc40b85cfe6.PNG)

This issue seem to persist regardless of whether I perform historical forecasts on a training dataset, validation dataset or an out of sample test-set.

My data is both multiseries and multivariate (multiple sensors recording in a minute frequency), and I encounter this same issue regardless of whether I use the TFT-model or the N-HiTS model. I use an input_chunk_length of 24 and an output_chunk_length of 6. 

Can anyone explain to me what might be the issue here? I currently use the TimeSeries.from_dataframe method. Should I instead use some form of shifted dataset to counterbalance the lag in predictions, and if so, could someone provide and example?

**Comment by pedronarloch** [comment]
Hey @nikobenho I saw you closed the topic. What was the solutions for your question?

**Comment by nikobenho** [comment]
> Hey @nikobenho I saw you closed the topic. What was the solutions for your question?

Hi @pedronarloch 
I closed the issue because I originally thought I had made a mistake using the historical forecasts method. As it turns out (as far as I can tell) the issue is not with Darts but rather that my problem is not best solved using the standard MAE loss function. This is the reason I closed the issue (I'm still experimenting with different loss functions and this seem to improve upon the results at least a little bit).
